server/middleware/headers.go

```diff
@@ -16,18 +16,35 @@ package middleware
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	api "github.com/tigrisdata/tigris/api/server/v1"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 var OutgoingHeaders = metadata.New(map[string]string{})
 
+const (
+	OutboundCookieHeader = "Set-Cookie"
+	CookieMaxAgeKey      = "Expires"
+)
+
 func headersUnaryServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		resp, err := handler(ctx, req)
+		callHeaders := metadata.New(map[string]string{})
 
-		if err := grpc.SendHeader(ctx, OutgoingHeaders); err != nil {
+		// add cookie header for sticky routing for interactive transactional operations
+		switch ty := resp.(type) {
+		case *api.BeginTransactionResponse:
+			{
+				expirationTime := time.Now().Add(time.Second * 10)
```

Comment:
Make this configurable?

Reply:
At lest define a constant. Config maybe an overkill.

Reply:
MaxTransactionTime+ some buffer in milliseconds should be sufficient here. The const of max transaction time is defined in timeout.go, please use that.

Reply:
👍 referred to existing constant + 5sec leeway.